### PR TITLE
Update to latest .Net SDK 7.0

### PR DESCRIPTION
**Pull request by kburtram**
Update to latest stable SDK for general product maintenance and health.

### CODE REVIEW

File: src/Microsoft.InsightsGenerator/DataTransformation.cs [hunk 1]
```diff
@@ -34,6 +34,7 @@ public DataArray Transform(DataArray array)
 
         private string[] GetColumnLabels(DataArray array, out DataArray.DataType[] columnDataType)
         {
+            #pragma warning disable CA1854
```
Comment (alanrenmsft) [inline src/Microsoft.InsightsGenerator/DataTransformation.cs:37]: comments explaining why?

**Review by alanrenmsft — APPROVED** [review]

File: src/Microsoft.InsightsGenerator/DataTransformation.cs (hunk 1 above)
Comment (kburtram) [reply src/Microsoft.InsightsGenerator/DataTransformation.cs:37]: sure, I can a comment.  The SDK update is bringing in a bunch of additional static analysis, that I'd rather fix incrementally than as part of the bump.  I'll also create an issue to track following up on reenabling the disabled rules.

**Comment by cheenamalhotra** [comment]
.NET 7 is not an LTS release.. do we not need an LTS version?
https://dotnet.microsoft.com/en-us/platform/support/policy

**Review by caohai — APPROVED** [review]

**Comment by kburtram** [comment]
> .NET 7 is not an LTS release.. do we not need an LTS version? https://dotnet.microsoft.com/en-us/platform/support/policy

@cheenamalhotra we generally update to each new major .Net SDK unless there is some reason not to.  Given this we don't worry about whether the version is LTS.

**Review by cheenamalhotra — APPROVED** [review]
LGTM, just eyeing into the test failures, cannot find an obvious reason... can we try rerunning?

**Pull request merged**